velox/common/fuzzer/ConstrainedGenerators.h

```diff
@@ -375,4 +388,52 @@ class JsonInputGenerator : public AbstractInputGenerator {
 
   folly::json::serialization_opts opts_;
 };
+
+/// Generates a JSON path for JSON of a given type.
+/// @param jsonType The type of data represented by the JSON.
+/// @param mapKeys Candidate key values of maps in the JSON. All maps in the
+/// JSON are assumed to share the same key type and candidate key values.
+/// @param maxContainerLength The maximum length of a container (array or map)
+/// in the JSON.
+/// @param makeRandomVariation If true, the generator will generate JSON path
+/// not supported by JsonExtractor.
+/// This generator generates the following JSON paths:
```

Comment:
the recursive operator `..` is now also supported. But we can always add it as a follow up.
Also, let add the ability to generated longer paths later like `$.key[index]..[index]`

Reply:
Hi @bikramSingh91, what's the syntax of recursive operator supported in JsonExtractor today? I locally made the generator to generate the following paths but JsonExtractor.cpp:61 throws "Invalid JSON path" error on them:
$..[2]
$..['k2']
$..k1
$..["k1"]

Reply:
The syntax is correct. Can you double check if you are on the latest?
My change adds a bunch of tests that should fail if this one is not working for you. You can take a look at the test examples here: https://github.com/facebookincubator/velox/pull/12568